rusk-wallet/src/main.rs

```diff
@@ -421,21 +421,24 @@ fn interactive(path: PathBuf) -> Result<LocalStore, Error> {
     }
 }
 
-/// Scan data directory and return a list of filenames
+/// Scan data directory and return a list of wallet names
 fn find_wallets(dir: &str) -> Result<Vec<String>, Error> {
-    // scan for wallets
     let dir = fs::read_dir(dir)?;
-    let names = dir
-        .map(|entry| {
-            entry
-                .ok()
-                .and_then(|e| {
-                    e.path()
-                        .file_name()
-                        .and_then(|name| name.to_str().map(String::from))
-                })
-                .unwrap()
-        })
-        .collect::<Vec<String>>();
-    Ok(names)
+
+    // filter files with '.dat' extension
+    let mut wallets: Vec<String> = Vec::new();
+    for el in dir {
```

Comment:
`dir` is an `Iterator`, meaning you can use actual `map` and `filter` calls to do what you're doing here and then collect at the end, which seems more appropriate.

Btw, what happens if there is a directory with a `.dat` extension?